packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ProjectController.java

```diff
@@ -523,6 +526,74 @@ public ResponseEntity<Project> updateProject(
 		return ResponseEntity.ok(updatedProject.get());
 	}
 
+	@Operation(summary = "Export a project")
+	@ApiResponses(
+			value = {
+				@ApiResponse(
+						responseCode = "200",
+						description = "The project export",
+						content = {
+							@Content(
+									mediaType = "application/json",
+									schema =
+											@io.swagger.v3.oas.annotations.media.Schema(
+													implementation = ProjectExport.class))
+						}),
+				@ApiResponse(
+						responseCode = "403",
+						description = "The current user does not have read privileges to this project",
+						content = @Content),
+				@ApiResponse(responseCode = "404", description = "Project could not be found", content = @Content),
+				@ApiResponse(
+						responseCode = "503",
+						description = "An error occurred when trying to communicate with either the postgres or spicedb"
+								+ " databases",
+						content = @Content)
+			})
+	@GetMapping("/export/{id}")
+	@Secured(Roles.USER)
+	public ResponseEntity<ProjectExport> exportProject(@PathVariable("id") final UUID id) {
+		projectService.checkPermissionCanRead(currentUserService.get().getId(), id);
+		try {
+			return ResponseEntity.ok(cloneService.exportProject(id));
```

Comment:
I do not understand where the resulting file from ProjectExport.clone is, or what is is?

Reply:
I don't think Kevin has written the UI to use the export feature, so in theory whatever call makes this request will receive a massive JSON blob, what they chose to do with it is up to the receiver - that is my reading of the code

Reply:
Yeah that's basically it, you will get a giant JSON blob containing everything needed to import. We can play around with the exact format, maybe use a .zip file instead rather than a .json file.